x/cronos/events/event.go

```diff
@@ -0,0 +1,133 @@
+package events
+
+import (
+	"fmt"
+	"strings"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+const maxIndexedArgs = 3
+
+// EventDescriptor describes how to convert an native event to a eth log
+type EventDescriptor struct {
+	id         common.Hash
+	indexed    []string
+	nonIndexed []string
+	packValues func([]interface{}) ([]byte, error)
+}
+
+func NewEventDescriptors(a abi.ABI) map[string]*EventDescriptor {
+	descriptors := make(map[string]*EventDescriptor, len(a.Events))
+	for _, event := range a.Events {
+		event_type := toUnderScore(event.Name)
+		descriptors[event_type] = &EventDescriptor{
+			id:         event.ID,
+			indexed:    getArguments(event.Inputs, true),
+			nonIndexed: getArguments(event.Inputs, false),
+			packValues: event.Inputs.NonIndexed().PackValues,
+		}
+	}
+	return descriptors
+}
+
+func makeFilter(
+	valueDecoders ValueDecoders,
+	attrs map[string]string,
+	attrNames []string) ([]any, error,
+) {
+	results := make([]any, 0, len(attrNames))
+	for _, name := range attrNames {
+		value, ok := attrs[name]
+		if !ok {
+			return nil, fmt.Errorf("attribute %s not found", name)
+		}
+		decode, ok := valueDecoders[name]
+		if !ok {
+			return nil, fmt.Errorf("value decoder for %s not found", name)
+		}
+		values, err := decode(value, true)
```

Comment:
is the `boolean` parameter different for index/non-index?

Reply:
yea, true for topic, false for data